Automattic/wp-calypso
# Third-Party Premium Themes: Adding CTA on themes page

#### Proposed Changes

* Add the appropriate CTAs for the third-party premium themes on the theme(`/theme/{themeId}/{siteSlug}`) page.
* Fixed the CSS for the CTA.

**Site with a Business plan**
<img width="757" alt="203050568-885e17ae-764c-487b-a89b-84cf99efd7d7" src="https://user-images.githubusercontent.com/1234758/203303319-7eec160c-52bb-449c-860a-04c9509d87c6.png">

**Site without a Business plan**
<img width="759" alt="Screen Shot 2022-11-21 at 09 00 46" src="https://user-images.githubusercontent.com/1234758/203303332-ba5b558b-79aa-49c6-8442-d29282f1a84f.png">

![Screen Shot 2022-11-22 at 10 58 13](https://user-images.githubusercontent.com/1234758/203334272-634116dc-b3b5-4366-94ff-22f3c321218b.png)


**User not authenticated**
<img width="756" alt="Screen Shot 2022-11-21 at 08 57 15" src="https://user-images.githubusercontent.com/1234758/203303348-8b6900a4-f8c6-4713-b403-7974b214cfb8.png">

#### Testing Instructions

Follow the instructions in this link D92217-code to sandbox the public-api.

1. Testing when the theme is not purchased but the user is eligible for externally managed themes.
1.1 Access `/theme/makoney/{SITE-SLUG}`, where the site has the business plan.
1.2 You should see the CTA inviting users to subscribe to the theme.
1.3 Clicking on the CTA should add the external-managed theme to the cart and redirect to the checkout page.
1. Testing when the theme is not purchased and the user is not eligible for externally managed themes.
2.1 Access `/theme/makoney/{SITE-SLUG}` with a free site.
2.2 You should see the CTA inviting users to upgrade.
2.3 Clicking on the CTA should add the external-managed theme, add the monthly Business plan and the theme to the cart, and redirect to the checkout page.

Related to #69468 #69480

Depends on:
 * #69832
 * #69852
 * #69831
 * #69853
 * #70214 
 * #70227

Closes #71083
